source/api/commands/route2.md

```diff
@@ -0,0 +1,385 @@
+
+Unlike {% url `cy.route()` route %}, `cy.route2()`:
+
+- can mock all types of network requests (i.e., Fetch API, page loads, XMLHttpRequests, etc.)
+- no longer requires {% url `cy.server()` server %} before use
+
+# Syntax
+
+```javascript
+cy.route2(requestMatcher)
+cy.route2(requestMatcher, response)
+cy.route2(method, requestMatcher)
+cy.route2(method, requestMatcher, response)
+```
+
+## Usage
+
+**{% fa fa-check-circle green %} Correct Usage**
+
+```javascript
+cy.route2('/users/**')
+```
+
+## Arguments
+
+**{% fa fa-angle-right %} requestMatcher** ***(String, Glob, RegExp, Object)***
+
+The `requestMatcher` argument allows you to listen for a route matching a specific URL or pattern.
+
+For simple route matching, passing a string, glob, or RegExp is the simplest method:
+
+```js
+// String
+cy.route2('/users)
+
+// Regex
+cy.route2(/users\/\d+/)
+
+// Glob to match all paths in the /user path
+cy.route2('/users/**')
+
+// https://localhost:8080/users/1b2c3            <-- matches
+// https://localhost:8080/users/profile/edit  <-- matches
+// https://localhost:8080/users/transaction?month=03&year=2020 <-- matches
+
+```
+
+However, if you need to match additional properties on the route, you can pass an object instead.
+
+```js
+cy.route2({
+  path: '/users/**',
+  method: 'POST',
+})
+```
+
+The following contains a complete list of available properties you can verify against:
+
+Option | Default | Description
+--- | --- | ---
+`auth` | `null` | Object with a `username` and `password` to match HTTP basic authentication
+`headers` | `null` | Object to match client request headers
+`hostname` | `null` | String, RegExp, or Glob to match based on requested hostname
+`method` | `'ALL'` | Match based on HTTP method (`GET`, `POST`, `PUT`, etc.)
+`path` | `null` | String, RegExp, or Glob to match on request path after the hostname, including query params
+`pathname` | `null` | String, RegExp, or Glob to match on request path after the hostname, without query params
+`port` | `null` | Match based on requested port number
+`query` | `null` | Object to match based on query parameters
+`url` | `null` | String, RegExp, or Glob to match based on full request URL
+
+**{% fa fa-angle-right %} response** ***(String, Object, Array, Function)***
+
+This allows you to supply a response `body` to {% url 'stub' stubs-spies-and-clocks#Stubs %} in the matching route. This is accomplished by giving you access to the request that you can use in a plain function.
+
+```js
+cy.route2('/users/**', (req) => {
+  req.send({
+    statusCode: 200,
+    body: JSON.stringify({
+      profile: {
+        firstName: 'Tony',
+        lastName: 'Jarvis'
+      }
+    })
+  })
+})
+```
+
+In addition, because you have access to the request, this means that you can modify the request as well:
+
+```js
+cy.route2('/users/**', (req) => {
+  req.headers = { ...req.headers, accept: 'application/json' }
+  req.body = { ...req.body, note: 'Custom note' }
```

Comment:
`req.body` will currently be a string in all situations, iirc. Might want to write a test to make sure, let me know if you want to pair on this

Reply:
still needs to be resolved